core/src/miner/backup.rs

```diff
@@ -16,10 +16,51 @@
 
 use super::mem_pool_types::MemPoolItem;
 use crate::db as dblib;
+use crate::error::Error;
+use crate::miner::mem_pool_types::TxOrigin;
+use crate::transaction::UnverifiedTransaction;
+use ckey::Error as KeyError;
+use ctypes::BlockNumber;
 use kvdb::{DBTransaction, KeyValueDB};
 use primitives::H256;
 use rlp::Encodable;
 use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+
+pub type PoolingInstant = BlockNumber;
+
+#[derive(Clone, Eq, PartialEq, Debug, RlpEncodable, RlpDecodable)]
+pub struct MemPoolItemProjection {
+    /// Transaction.
+    pub tx: UnverifiedTransaction,
+    /// Transaction origin.
+    pub origin: TxOrigin,
+    /// Insertion time
+    pub inserted_block_number: PoolingInstant,
+    /// Insertion timstamp
+    pub inserted_timestamp: u64,
+    /// ID assigned upon insertion, should be unique.
+    pub insertion_id: u64,
+}
+
+impl TryFrom<MemPoolItemProjection> for MemPoolItem {
+    type Error = Error;
+    fn try_from(mem_pool: MemPoolItemProjection) -> Result<Self, Error> {
+        let tx = mem_pool.tx;
+        if tx.verify_transaction() {
```

Comment:
Do we need to double check the content of the `UnverifiedTransaction`? tx.verify_transaction() will be called inside of the `tx.try_into()`. Am I missing something?

Reply:
What do you think about changing `verify_transaction` method to be `private`? Only `try_into` is enough to be public because verification will only be conducted to convert unverified one into verified one.

Reply:
If we change `verified_transaction()` to `private method`, we can use it on `VerfiedTransaction`'s `Impl` because `verified_transaction()` is implemented for `UnverifiedTransaction`. So, I didn't modified it.
However, double-check is my mistake, so I modified it. Thank you for a good review.